### DISCUSSION TITLE
是否已经支持类似于buildroot、openwrt的组件包构建

### DISCUSSION THREAD

**Discussion by justa-cai** [xmake-io/xmake#1623] [General]
### 你在什么场景下需要该功能？

1. 组件构建，从所有的源码构建生成二进制文件

### 描述可能的解决方案

1. 创建项目
    xmake project create 
2. 选择组件
    xmake project menu
3. 构建
    xmake build
4. 选择构建指定组件
   xmake {module_name}/{dir_name}

### 描述你认为的候选方案

无

### 其他信息

buildroot、openwrt的packege仓库很详细，可以引用

**Comment by waruqi** [comment]
你需要自己配置 xmake.lua 。。menu 特性也支持 xmake f --menu 

组件选项选择，自己配置 option ，如果编译源码，如何生成库到指定目录，还是需要自己配置

想要一键构建buildroot 完整特性，目前没有 example ，理论上通过 xmake.lua 自己配置搞，肯定全能做，但是需要用户自己弄。

**Comment by xq114** [comment]
xmake目前还不支持编译管理linux内核、gcc、cross toolchain等，目前xmake专注于project级别的构建支持而不是操作系统的全局包管理。组件需要在xmake.lua文件中动态添加、配置。buildroot/openwrt这类属于定制化linux系统了，目前xmake与它们的功能并不重叠，完全可以同时使用，先使用buildroot/openwrt构建系统，其中这两个没有提供，或者需要 指定版本/多版本、多工具链并存/其他xmake特有功能 的包再用xmake管理。当然也不排除未来xmake增加编译环境/toolchain的管理功能

**Comment by waruqi** [comment]
其实等 xmake-repo 仓库里面支持交叉编译的包 到一定量级后，单独基于 xmake.lua 配置和包管理做个buildroot 项目出来，也是很方便的，目前toolchain也基本可配置，menu/option也都ok

之前rthread的人也跟我提过类似需求，关键还是没人力，没时间搞 等回头看看吧 说不定哪天我兴致来了 搞一个也是有可能的
